docs/src/further_topics/ugrid/operations.rst

```diff
@@ -0,0 +1,953 @@
+
+        >>> face_x = AuxCoord([2.0, 6.0], "longitude")
+        >>> face_y = AuxCoord([1.0, 1.0], "latitude")
+
+        >>> edge_node_c = Connectivity(
+        ...     indices=[[0, 1], [0, 2], [1, 3], [1, 4], [2, 3], [3, 4]],
+        ...     cf_role="edge_node_connectivity",
+        ...     attributes={"demo": "Supports every standard CF property"},
+        ... )
+
+        # Create some dead-centre edge coordinates.
+        >>> edge_x, edge_y = [
+        ...     AuxCoord(
+        ...         node_coord.points[edge_node_c.indices_by_location()].mean(axis=1),
+        ...         node_coord.standard_name,
+        ...     )
+        ...     for node_coord in (node_x, node_y)
+        ... ]
+
+        >>> face_indices = np.ma.masked_equal([[0, 1, 3, 2], [1, 4, 3, 999]], 999)
+        >>> face_node_c = Connectivity(
+        ...     indices=face_indices, cf_role="face_node_connectivity"
+        ... )
+
+        >>> my_mesh = Mesh(
+        ...     long_name="my_mesh",
+        ...     topology_dimension=2,  # Supports 2D (face) elements.
+        ...     node_coords_and_axes=[(node_x, "x"), (node_y, "y")],
+        ...     connectivities=[edge_node_c, face_node_c],
+        ...     edge_coords_and_axes=[(edge_x, "x"), (edge_y, "y")],
+        ...     face_coords_and_axes=[(face_x, "x"), (face_y, "y")],
+        ... )
+
+.. _making a cube:
+
+Making a Cube (with a Mesh)
+---------------------------
+.. |tagline: making a cube| replace:: |unchanged|
+
+.. rubric:: |tagline: making a cube|
+
+Creating a :class:`~iris.cube.Cube` is unchanged; the
+:class:`~iris.experimental.ugrid.Mesh` is linked via a
+:class:`~iris.experimental.ugrid.MeshCoord` (see :ref:`ugrid MeshCoords`):
+
+.. dropdown:: :opticon:`code`
+
+    .. doctest:: ugrid_operations
+
+        >>> import numpy as np
+
+        >>> from iris.coords import DimCoord
+        >>> from iris.cube import Cube, CubeList
+
+        >>> vertical_levels = DimCoord([0, 1, 2], "height")
+
+        >>> my_cubelist = CubeList()
+        >>> for conn in (edge_node_c, face_node_c):
+        ...    location = conn.location
+        ...    mesh_coord_x, mesh_coord_y = my_mesh.to_MeshCoords(location)
+        ...    data_shape = (len(conn.indices_by_location()), len(vertical_levels.points))
+        ...    data_array = np.arange(np.prod(data_shape)).reshape(data_shape)
+        ...
+        ...    my_cubelist.append(
+        ...        Cube(
+        ...            data=data_array,
+        ...            long_name=f"{location}_data",
+        ...            units="K",
+        ...            dim_coords_and_dims=[(vertical_levels, 1)],
+        ...            aux_coords_and_dims=[(mesh_coord_x, 0), (mesh_coord_y, 0)],
+        ...        )
+        ...    )
+
+        >>> print(my_cubelist)
+        0: edge_data / (K)                     (-- : 6; height: 3)
+        1: face_data / (K)                     (-- : 2; height: 3)
+
```

Comment:
Is it worth also printing a cube at this point, to show how MeshCoords map to the dims?

Reply:
d2e504f66a1f46335ae441c9494956ffb2aafb09